internal/keystore/azure/key-vault.go

```diff
@@ -216,51 +217,23 @@ func (s *Store) Delete(ctx context.Context, name string) error {
 	if err != nil {
 		return fmt.Errorf("azure: failed to delete '%s': %v", name, err)
 	}
-	if stat.StatusCode != http.StatusNotFound {
```

Comment:
I think there should hold this.

Reply:
Not sure what you mean. The new code is resilient to cases where a non-existing key is being purged.